functions/Get-DbaProcess.ps1

```diff
@@ -0,0 +1,168 @@
+
+.EXAMPLE   
+Get-DbaProcess -SqlServer sqlserver2014a -SqlCredential $credential -Spids 56, 77
+	
+Shows information about the processes for spid 56 and 57. Uses alternative (SQL or Windows) credentials to login to sqlserver2014a.
+
+.EXAMPLE   
+Get-DbaProcess -SqlServer sqlserver2014a -Programs 'Microsoft SQL Server Management Studio'
+	
+Shows information about the processes that were created in Microsoft SQL Server Management Studio.
+
+.EXAMPLE   
+Get-DbaProcess -SqlServer sqlserver2014a -Hosts workstationx, server100
+	
+Shows information about the processes that were initiated by hosts (computers/clients) workstationx and server 1000.
+	
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[object]$SqlServer,
+		[object]$SqlCredential,
+		[switch]$NoSystemSpids,
+		[switch]$Detailed
+	)
+	
+	DynamicParam { if ($sqlserver) { Get-ParamSqlAllProcessInfo -SqlServer $sqlserver -SqlCredential $SqlCredential } }
+	
+	BEGIN
+	{
+		$sourceserver = Connect-SqlServer -SqlServer $sqlserver -SqlCredential $SqlCredential
+		$source = $sourceserver.DomainInstanceName
+		
+		$logins = $psboundparameters.Logins
+		$spids = $psboundparameters.Spids
+		$exclude = $psboundparameters.Exclude
+		$hosts = $psboundparameters.Hosts
+		$programs = $psboundparameters.Programs
+		$databases = $psboundparameters.Databases
+	}
+	
+	PROCESS
+	{
+		$allsessions = @()
+		
+		$processes = $sourceserver.EnumProcesses()
+		
+		if ($nosystemspids -eq $true)
+		{
+			$allsessions += $processes | Where-Object { $_.spid -gt 50 }
+		}
+		
+		if ($logins.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Login -in $Logins }
+		}
+		
+		if ($spids.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Spid -in $spids }
+		}
+		
+		if ($hosts.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Host -in $hosts }
+		}
+		
+		if ($programs.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Program -in $programs }
+		}
+		
+		if ($databases.count -gt 0)
+		{
+			$allsessions += $processes | Where-Object { $_.Database -in $databases }
+		}
+		
+		if ($allsessions.urn.count -eq 0)
+		{
+			$allsessions = $processes
+		}
+		
+		if ($exclude.count -gt 0)
+		{
+			$allsessions = $allsessions | Where-Object { $exclude -notcontains $_.Spid }
+		}
+		
+		$duplicates = @()
+		
+		foreach ($session in $allsessions)
+		{
+			if ($session.spid -in $duplicates) { continue }
+			$duplicates += $session.spid
+			
+			$spid = $session.spid
+			$login = $session.login
+			$database = $session.database
+			$program = $session.program
+			$host = $session.host
+			
+		}
+	}
+	
+	END
+	{
+		$sourceserver.ConnectionContext.Disconnect()
+		
+		if ($Detailed)
+		{
+			return $allsessions | Select-Object Spid, Login, Host, Status, Command, Database, Cpu, MemUsage, BlockingSpid, IsSystem, Program
```

Comment:
EnumPorcesses() don't display info on `Status `and `Command` when query is sleeping. 
On sp_who2 this shows `'sleeping'` and `'AWAITING COMMAND'`

![image](https://cloud.githubusercontent.com/assets/19521315/19735383/3f27b9e0-9ba3-11e6-9e39-a04a12069962.png)

```
if ($Detailed)
		{
			return $allsessions | Select-Object Spid, Login, Host, Database, BlockingSpid, Program, @{ name = "Status"; expression = {if ($_.Status -eq "") {"sleeping"} else {$_.Status}} },  @{ name = "Command"; expression = {if ($_.Command -eq "") {"AWAITING COMMAND"} else {$_.Command}} }, Cpu, MemUsage, IsSystem
		}

		return ($allsessions | Select-Object Spid, Login, Host, Database, BlockingSpid, Program, @{ name = "Status"; expression = {if ($_.Status -eq "") {"sleeping"} else {$_.Status}} },  @{ name = "Command"; expression = {if ($_.Command -eq "") {"AWAITING COMMAND"} else {$_.Command}} })
```
This way, we keep it 100% SMO :smile: 

I also have changed columns' order (OCD strikes! :see_no_evil: )